lib/grammers-client/src/types/inline_query.rs

```diff
@@ -0,0 +1,169 @@
+use grammers_tl_types as tl;
+
+/// Represents an inline query update, which occurs when you sign in as a bot and a user sends an
+/// inline query such as `@bot query`.
+pub struct InlineQuery {
+    query: tl::types::UpdateBotInlineQuery,
+    client: Client,
+}
+
+/// An inline query answer builder.
+pub struct Answer {
+    request: tl::functions::messages::SetInlineBotResults,
+    client: Client,
+}
+
+impl InlineQuery {
+    pub(crate) fn new(client: &Client, query: tl::types::UpdateBotInlineQuery) -> Self {
+        Self {
+            query,
+            client: client.clone(),
+        }
+    }
+
+    // User that sent the query.
+    pub fn user_id(&self) -> i32 {
+        self.query.user_id
+    }
+
+    // The text of the inline query.
+    pub fn text(&self) -> &str {
+        self.query.query.as_str()
+    }
+
+    /// Answer the inline query.
+    // TODO: add example
+    pub fn answer(&self, results: Vec<tl::enums::InputBotInlineResult>) -> Answer {
+        Answer {
+            request: tl::functions::messages::SetInlineBotResults {
+                gallery: false,
+                private: false,
+                query_id: self.query.query_id,
+                results,
+                cache_time: 0,
+                next_offset: None,
+                switch_pm: None,
+            },
+            client: self.client.clone(),
+        }
+    }
+}
+
+impl Answer {
+    /// Whether the results should show as a gallery (grid) or not.
+    /// Defaults to false.
+    pub fn gallery(mut self, gallery: bool) -> Self {
+        self.request.gallery = gallery;
+        self
+    }
+
+    /// Whether the results should be cached by Telegram (not private) or by the user's client
+    /// (private).
+    /// Defaults to false.
+    pub fn private(mut self, private: bool) -> Self {
+        self.request.private = private;
+        self
+    }
+
+    /// For how long this result should be cached on the user's client. Defaults to 0 for no
+    /// cache.
+    pub fn cache_time(mut self, cache_time: i32) -> Self {
+        self.request.cache_time = cache_time;
+        self
+    }
+
+    /// The offset the client will send when the user scrolls the results and it repeats the
+    /// request.
+    pub fn next_offset(mut self, next_offset: impl Into<String>) -> Self {
+        self.request.next_offset = Some(next_offset.into());
+        self
+    }
+
+    /// If set, this text will be shown in the results to allow the user to switch to private
+    /// messages.
+    pub fn switch_pm(mut self, text: impl Into<String>, start_param: impl Into<String>) -> Self {
+        self.request.switch_pm = Some(tl::enums::InlineBotSwitchPm::Pm(
+            tl::types::InlineBotSwitchPm {
+                text: text.into(),
+                start_param: start_param.into(),
+            },
+        ));
+        self
+    }
+
+    /// Answers the inline query with the given results.
+    pub async fn send(self) -> Result<(), InvocationError> {
```

Comment:
I'd like to be able to `await` on the `Answer` directly, much like #72, but I guess we can do that after merging this with `send`.